src/core/parse.js

```diff
@@ -8,52 +8,55 @@ export let parse = (obj, selector) => {
-                          return sel ? sel + ' ' + k : k;
-                      });
-                  })
-                : key;
-
-            // If these are the `@` rule
-            if (key[0] == '@') {
-                // Handling the `@font-face` where the
-                // block doesn't need the brackets wrapped
-                if (key[1] == 'f') {
-                    blocks += parse(val, key);
+        // If this isn't an empty rule
+        if (val != undefined && val != null) {
+            // If this is a 'block'
+            if (typeof val == 'object') {
+                // If these are the `@` rule
+                if (key[0] == '@') {
+                    // Handling the `@font-face` where the
+                    // block doesn't need the brackets wrapped
+                    if (key[1] == 'f') {
+                        blocks += parse(val, key);
+                    } else {
+                        // Regular rule block
+                        blocks += key + '{' + parse(val, key[1] == 'k' ? '' : selector) + '}';
+                    }
                 } else {
-                    // Regular rule block
-                    blocks += key + '{' + parse(val, key[1] == 'k' ? '' : selector) + '}';
+                    // Call the parse for this block
+                    blocks += parse(
+                        val,
+                        selector
+                            ? // Go over the selector and replace the matching multiple selectors if any
+                              selector.replace(/([^,])+/g, (sel) => {
+                                  // Return the current selector with the key matching multiple selectors if any
+                                  return key.replace(/(^:.*)|([^,])+/g, (k) => {
+                                      // If the current `k`(key) has a nested selector replace it
+                                      if (/&/.test(k)) return k.replace(/&/g, sel);
+
+                                      // If there's a current selector concat it
+                                      return sel ? sel + ' ' + k : k;
+                                  });
+                              })
+                            : key
+                    );
                 }
             } else {
-                // Call the parse for this block
-                blocks += parse(val, next);
-            }
-        } else {
-            if (key[0] == '@' && key[1] == 'i') {
-                outer = key + ' ' + val + ';';
-            } else {
-                key = key.replace(/[A-Z]/g, '-$&').toLowerCase();
-                // Push the line for this property
-                current += parse.p
-                    ? // We have a prefixer and we need to run this through that
-                      parse.p(key, val)
-                    : // Nope no prefixer just append it
-                      key + ':' + val + ';';
+                if (/^@i/.test(key)) {
```

Comment:
Good refactor! You changed this to make it more compact or was there a size constraint? If I remember correctly I expanded the check into two to gain some more gzip size. But I might remember it wrong.

Reply:
I don't know? Can you test it or how can i test it?

Reply:
Now It's fixed